python/cudf/cudf/io/parquet.py

```diff
@@ -886,8 +896,9 @@ def _get_groups_and_offsets(
         grouped_df.reset_index(drop=True, inplace=True)
     grouped_df.drop(columns=partition_cols, inplace=True)
     # Copy the entire keys df in one operation rather than using iloc
-    part_names = part_keys.to_pandas().unique().to_frame(index=False)
-
+    part_names = (
+        part_keys.to_pandas(nullable=True).unique().to_frame(index=False)
+    )
```

Comment:
This is a good point. It's a shame to be using pandas to perform `unique`, but we need to preserve the initial order (including nulls), and we cannot rely on `Column.unique`, because `part_keys` may be a `MultiIndex` here. The good news is that we *should* be able to do `part_keys.take(part_offsets[:-1])` instead of `unique` anyway.

Reply:
Assuming the number of partition keys is small, I don' think pandas usage is really a performance footgun...